src/backend/distributed/README.md

```diff
@@ -0,0 +1,2079 @@
+
+Through these hooks, Citus can intercept any interaction between the client and the PostgreSQL engine that involves Citus tables. Citus can then replace or augment PostgreSQL's behavior. 
+
+# Query planner 
+
+Citus’ planner layered planner accommodates different workloads. There are several useful presentations/papers that are relevant to Citus’ distributed planner, below we try to categorize them: 
+
+## High-level design/flow: 
+
+- Distributing Queries the Citus Way: Marco’s PG Con presentation is almost up to date and can provide a good introduction. https://postgresconf.org/system/events/document/000/000/233/Distributing_Queries_the_Citus_Way.pdf
+
+- Another useful content on this topic is the Planner README.md: https://github.com/citusdata/citus/blob/main/src/backend/distributed/planner/README.md
+
+- Onder’s talk at CitusCon: https://www.youtube.com/watch?v=raw3Pwv0jb8 
+- Citus paper: https://dl.acm.org/doi/pdf/10.1145/3448016.3457551 
+
+- Logical planner design - 1: https://speakerdeck.com/marcocitus/scaling-out-postgre-sql 
+
+- Logical Planner design - 2: https://www.youtube.com/watch?v=xJghcPs0ibQ
+
+- Logical Planner based on the paper: Correctness of query execution strategies in distributed databases: https://dl.acm.org/doi/pdf/10.1145/319996.320009
+
+##  Distributed Query Planning with Examples in Citus (as of Citus 12.1) 
+   
+This part of the documentation aims to provide a comprehensive understanding of how Citus handles distributed query planning with examples. We will use a set of realistic tables to demonstrate various queries. Through these examples, we hope to offer a step-by-step guide on how Citus chooses to plan queries. 
+ 
+**Non-Goal:** 
+This section does not aim to provide the high level design of the distributed planner, but instead the lowest levels of details possible. For high-level flow, go to https://postgresconf.org/system/events/document/000/000/233/Distributing_Queries_the_Citus_Way.pdf 
+   
+ 
+ 
+ 
+### Table definitions used in this section 
+ 
+```sql 
+-- Distributed Table: Users Table 
+CREATE TABLE users_table ( 
+    user_id bigserial PRIMARY KEY, 
+    username VARCHAR(50) NOT NULL, 
+    email VARCHAR(50), 
+    date_of_birth DATE, 
+    country_code VARCHAR(3) 
+); 
+SELECT create_distributed_table('users_table', 'user_id'); 
+ 
+-- Distributed Table: Orders Table 
+CREATE TABLE orders_table ( 
+    order_id bigserial, 
+    user_id BIGINT REFERENCES users_table(user_id), 
+    product_id BIGINT, 
+    order_date TIMESTAMPTZ, 
+    status VARCHAR(20) 
+); 
+SELECT create_distributed_table('orders_table', 'user_id'); 
+ 
+ 
+-- Distributed Table: Products Table 
+CREATE TABLE products_table ( 
+    product_id bigserial PRIMARY KEY, 
+    product_name VARCHAR(100), 
+    category_id INT, 
+    price NUMERIC(10, 2) 
+); 
+SELECT create_distributed_table('products_table', 'product_id'); 
```

Comment:
Not relevant to the example and not a feedback. I'm just curious about whether this call will cause the products_table to be co-located with the other two distributed tables (which might not be desirable).